### PR TITLE
 add forcedly_disable_pmu 

### PR DESCRIPTION
**Pull request by justin-he**

### CODE REVIEW

File: cli/config/configuration-qemu-virtiofs.toml.in [hunk 1]
```diff
@@ -149,6 +149,11 @@ block_device_driver = "@DEFBLOCKSTORAGEDRIVER_QEMU@"
 # Default false
 #block_device_cache_noflush = true
 
+# Specifies pmu options for hvpervisor.
+# Denotes whether add pmu=off for guest
+# Default false
+#forcedly_disable_pmu = false
```
Comment (grahamwhaley) [inline cli/config/configuration-qemu-virtiofs.toml.in]: s/forcedly/forcibly/ would read better.

**Comment by grahamwhaley** [comment]
From Travis CI - I presume this is related:

```
pkg/katautils/config.go:86:17: struct of size 376 bytes could be of size 368 bytes (maligned)
type hypervisor struct {
                ^
The command ".ci/static-checks.sh" failed and exited with 1 during .
```

**Comment by grahamwhaley** [comment]
I should also say - nice - I'm glad to see work on reducing boot time :-)
Please also ensure any relevant documents are updated or created - we must document all our tweakables clearly.

**Comment by devimc** [comment]
Thanks @justin-he for raising this, I'll change this to `do-not-merge` because I'd like to discuss something fist.
I like the idea of having an option/annotation to disable/enable CPU features, for example pmu, vmx, etc,  but same as the machine accelerators [1], I consider that the best approach is to let users specify them. How about adding a new option in the configuration file (something like `cpu_features` ? ) to let users enable/disable CPU features?


[1] - https://github.com/kata-containers/runtime/blob/master/cli/config/configuration-qemu.toml.in#L36

**Comment by amshinde** [comment]
I am inclined to make this option as enabled by default, as long as we document this, and let users know that this needs to be turned on for doing any perf measurements.

**Comment by c3d** [comment]
@devimc I like the option.

This raises an interesting question in a k8s context. There should be a way to automatically get labels for such options, so that you could select pods which run this or that option.

**Comment by justin-he** [comment]
tend to close it, And I will open another new issue to follow up adding cpu_features.

**Comment by justin-he** [comment]
https://github.com/kata-containers/runtime/issues/2576 was opened

**Pull request closed without merging**